src/lib.rs

```diff
@@ -2182,18 +2204,8 @@ impl Build {
                     cmd.push_cc_arg(format!("-stdlib=lib{}", stdlib).into());
                 }
                 _ => {
-                    let stdout = io::stdout();
-                    {
-                        let mut stdout = stdout.lock();
-                        if self.cargo_warnings {
-                            let _ = write!(stdout, "cargo:warning=");
-                        }
-                        let _ = writeln!(
-                            stdout,
-                            "cpp_set_stdlib is specified, but the {:?} compiler \
-                         does not support this option, ignored",
-                            cmd.family,
-                        );
+                    if self.cargo_warnings {
+                        println_warning!("cpp_set_stdlib is specified, but the {:?} compiler does not support this option, ignored", cmd.family);
                     }
```

Comment:
Sorry for the confusion, but I was actually thinking of extracting this as a function and macro:

```rust
impl Build {
    fn println_warning(&self, arg: &fmt::Argument<'_>) {
        if self.cargo_warnings {
            println!("cargo:warning={}", arg);
        }
    }
}

macro_rules! println_warning {
    ($this:expr, $s:literal) => {
        this.println_warning(format_args("{}", s))
    };
    ($dst:expr, $fmt:literal, $($arg:tt)*) => {
        this.println_warning(format_args!($fmt, $($arg)*)))
    };
}
```

Reply:
Yes, sorry for the confusion.

Reply:
> I see the appeal of having this println_warning method mirroring the design of the print method, but that means that the macro can only be used when you have access to Build.

I think having a `Build::println_warning` that accepts `& dyn Display`, user can pass in `&str` or `&fmt::Arguments<'_>`:

```
impl Build {
    fn println_warning(&self, arg: &dyn Display) {
        if self.cargo_warnings {
            println!("cargo:warning={}", arg);
        }
    }
}


```

Reply:
Makes sense. Sorry for the confusion on my part :)